tests/test_hls_report.py

```diff
@@ -0,0 +1,207 @@
+    s[sY].pipeline(sY.axis[1])
+    s[sR].pipeline(sR.axis[1])
+    s[sF].pipeline(sF.axis[1])
+  
+    target = hcl.platform.zc706  
+    target.config(compile="vivado_hls", mode="csyn")
+  
+    hcl_img = hcl.asarray(img)
+    hcl_Gx = hcl.asarray(np.array([[-1,-2,-1],[0,0,0],[1,2,1]]))
+    hcl_Gy = hcl.asarray(np.array([[-1,0,1],[-2,0,2],[-1,0,1]]))
+    hcl_F = hcl.asarray(np.zeros((height, width)))
+  
+    f = hcl.build(s,target) 
+    f(hcl_img, hcl_Gx, hcl_Gy, hcl_F)
+    return f.report()
+
+def parse_rpt():
+    path = pathlib.Path(__file__).parent.absolute()
+    xml_file = str(path) + '/test_report_data/test_csynth.xml'
+    with open(xml_file, "r") as xml:
+        profile = xmltodict.parse(xml.read())["profile"]
+    clock_unit = profile["PerformanceEstimates"]["SummaryOfOverallLatency"]["unit"]
+    summary = profile["PerformanceEstimates"]["SummaryOfLoopLatency"]
+  
+    info_table = hcl.report.Displayer(clock_unit)
+    info_table.scan_range(summary)
+    info_table.get_loops(summary)
+    info_table.init_data(summary)
+    return info_table
+
+def refine(lst):
+    pattern = re.compile(r'\s\s+') 
+    res = []
+    for i in lst:
+        res.append(re.sub(pattern, ', ', i))
+    res[0] = res[0].replace(', ', '', 1)
+    return res
+
+def get_expected(wd):
+    path = pathlib.Path(__file__).parent.absolute()
+    path = str(path) + '/test_report_data/expected.json'
+    with open(path) as f:
+        data = json.loads(f.read())
+    return data[wd]
+
+def test_info(vhls):
+    if vhls:
+        rpt = sobel()
+    else:
+        rpt = parse_rpt()
+    res = rpt.display()
+    res_str = res.split("\n") 
+    lst = refine(res_str)
+    assert lst == get_expected("NoQuery")
+
+def test_loop_query(vhls):
+    if vhls:
+        rpt = sobel()
+    else:
+        rpt = parse_rpt()
+    row_query = ['P', 'A']
+    lq = rpt.display(loops=row_query)
+    lq_str = lq.split("\n") 
+    lq_lst = refine(lq_str)
+    assert lq_lst == get_expected("LoopQuery")
+
+def test_column_query(vhls):
+    if vhls:
+        rpt = sobel()
+    else:
+        rpt = parse_rpt()
+    col_query = ['Latency']
+    cq = rpt.display(cols=col_query)
+    cq_str = cq.split("\n")
+    cq_lst = refine(cq_str)
+    assert cq_lst == get_expected("ColumnQuery")
+
+def test_level_query(vhls):
+    if vhls:  
+        rpt = sobel()
+    else:
+        rpt = parse_rpt()
+    lev_query = 1
+    vq = rpt.display(level=lev_query)
+    vq_str = vq.split("\n")
+    vq_lst = refine(vq_str)
+    assert vq_lst == get_expected("LevelQuery")
+
+def test_level_oob_query(vhls):
+    if vhls:
+        rpt = sobel()
+    else:
+        rpt = parse_rpt()
+    lev_query = 5
+    vq = rpt.display(level=lev_query)
+    vq_str = vq.split("\n")
+    vq_lst = refine(vq_str)
+    assert vq_lst == get_expected("LevelQueryOOB")
+    lev_query = -2
+    try:
+        vq = rpt.display(level=lev_query)
+        assert False
+    except IndexError:
+        assert True
```

Comment:
```suggestion
    try:
        vq = rpt.display(level=lev_query)
    except IndexError:
        assert True
    assert False
```
This also handles the case when the exception is not `IndexError`.

Reply:
But if the control flow is modified that way, doesn't `assert False` always get executed in cases where `IndexError` occurs? I think adding one more `except` branch that captures the other types of errors would be better.

Reply:
Sorry, my changes were not complete. We should replace `assert True` with `return`.